crates/common/tedge_utils/src/fs_notify.rs

```diff
@@ -0,0 +1,600 @@
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    path::{Path, PathBuf},
+};
+
+use async_stream::try_stream;
+pub use futures::{pin_mut, Stream, StreamExt};
+use inotify::{EventMask, Inotify, WatchMask};
+use maplit::{btreeset, hashmap};
+use strum_macros::Display;
+use try_traits::default::TryDefault;
+
+const DIRECTORY_IS_WATCHED: &str = "C8Y_DIR_WATCH";
+
+#[derive(Debug, Display, PartialEq, Eq, Clone, Hash, PartialOrd, Ord)]
```

Comment:
Why do we need the `PartialOrd` and `Ord` implementations? Can these variants be ordered?

Reply:
I do the `PartialOrd`, `Ord` to temporarily insert the enum into a BTreeSet (ordering them) and then convert them to a Vec<FileEvent> to end up with an ordered vector. Is there a better way to achieve a sorted vector?  :sweat_smile:  

Thanks for the hint to derive `Copy`.

Reply:
> Is there a better way to achieve a sorted vector?

Yes, `Vec` derefs to slice, which has [`sort()`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.sort)!